src/crypto/sha512.cpp

```diff
@@ -205,3 +212,43 @@ CSHA512& CSHA512::Reset()
     sha512::Initialize(s);
     return *this;
 }
+
+int main(){
+
+
+    //Prepare test data
+    unsigned char buf[64] = {0};
```

Comment:
Could be wrong (sorry I've just broken my VM or I'd test), but... you've got an `unsigned char` here, and `char` below. Could setting them to FF be causing any sort of difference between signed/unsigned?

Reply:
No, the underlying data is the same, that really becomes an issue when you interpret that data, through types usually; comparing for example, as big positive numbers become negative. I will test nontheless.

Reply:
Same result.